### PR TITLE
feat(api): DELETE /user/ms-username

### LINKED ISSUE #51429: DELETE /user/ms-username

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes https://github.com/freeCodeCamp/freeCodeCamp/issues/51429

It would be less hassle if https://github.com/freeCodeCamp/freeCodeCamp/pull/51776 went in first, but it's not a big deal.

### CODE REVIEW

**Comment by naomi-lgbt** [comment]
51776 is in - mind resolving the conflicts?

File: api/src/routes/user.test.ts [hunk 1]
```diff
@@ -301,6 +358,32 @@ describe('userRoutes', () => {
...
+        await fastifyTestInstance.prisma.userToken.createMany({
+          data: tokenData
+        });
+
+        await superRequest('/account/reset-progress', {
+          method: 'POST',
+          setCookies
+        });
+
+        expect(await fastifyTestInstance.prisma.userToken.count()).toBe(1);
```
Comment (moT01) [inline api/src/routes/user.test.ts:385]: Wonder if we should check that the remaining item is the one that's expected on these tests that use the count.

**Review by moT01 — COMMENTED** [review]
This mostly LGTM :tada: I left one thought. 

Going to close and open to make the node 20 tests run.

**Comment by Sembauke** [comment]
@ojeytonwilliams sorry for not reviewing this sooner. Could you resolve the conflicts again?

**Comment by ojeytonwilliams** [comment]
No worries, @Sembauke. I think it's ready for review again.

**Review by Sembauke — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Pull request merged**